kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/ClientSaslContextImpl.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.internal;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import io.kroxylicious.proxy.authentication.ClientSaslContext;
+
+import edu.umd.cs.findbugs.annotations.DefaultAnnotation;
+import edu.umd.cs.findbugs.annotations.DefaultAnnotationForParameters;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import edu.umd.cs.findbugs.annotations.ReturnValuesAreNonnullByDefault;
+
+@ReturnValuesAreNonnullByDefault
+@DefaultAnnotationForParameters(NonNull.class)
+@DefaultAnnotation(NonNull.class)
+public class ClientSaslContextImpl implements ClientSaslContext {
+
+    private @Nullable String clientAuthorizationId;
+    private @Nullable String mechanism;
+    private @Nullable String proxyServerId;
```

Comment:
wonder if you'd be better off with a single field using an inner record like:

`private record Authorized(String clientAuthorizationId, String mechanism, String proxyServerId, @Nullable String proxyServerId) implements ClientSaslContext`

so you swap all 3 with a single assignment between `null` and an `Authorized` instance and return an `Optional.ofNullable(x)` from `clientSaslContext()`. Would remove the requireNonNulls and this mutable thing wouldn't implement CSC

Reply:
Thanks.